### DISCUSSION TITLE
Filters data

### DISCUSSION THREAD

**Discussion by zeeshantariq08** [filamentphp/filament#9768] [Help]
how can i get the data in a filter if we don't have a data in a table query .

protected function getTableQuery(): ?\Illuminate\Database\Eloquent\Builder
    {
        return  MyQ::query()
            ->with('workflow_status')
            ->where('assignee_id', auth()->user()?->id)
            ->whereHas('workflow_status.status', function ($query) {
                // Exclude records with the "Approved" status (assuming ID 2)
                $query->where('id', '!=', 8);
            });

}
but on filter, i want data which are approved

**Comment by leandrocfe** [comment]
Sorry, I didn't understand what you need

**Reply by zeeshantariq08** [reply]
i am fetching the data using getTableQuery and the data will not include which has status id 8.so data is coming .
when i use filter on status , i want that if i sent status id 8 it will show a record but it is not showing . my filter code is

 Tables\Filters\Filter::make('workflow_status')->form([
                    Forms\Components\Select::make('status_id')->options(function () {
                        return Status::pluck('name', 'id');
                    }
                    )->label('Status')->searchable(),
                ])->query(function (Builder $query, array $data): Builder {
                    return $query
                        ->when(
                            $data['status_id'],
                            fn(Builder $query, $status_id): Builder => $query->whereHas('workflow_status', fn(Builder $query) => $query->where('workflow_status_id', $status_id))
                        );
                }),
